apps/web/src/routes/UserSettings/NotificationSettingsTab/NotificationSettingsTab.js

```diff
@@ -1,255 +1,212 @@
-import React, { Component } from 'react'
-import { useTranslation, withTranslation } from 'react-i18next'
-import { compact } from 'lodash/fp'
-import PropTypes from 'prop-types'
+import { includes, every } from 'lodash/fp'
+import React, { useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useDispatch, useSelector } from 'react-redux'
+import { createSelector } from 'reselect'
 import Tooltip from 'components/Tooltip'
 import Icon from 'components/Icon'
 import Loading from 'components/Loading'
-import Select from 'components/Select'
-import { bgImageStyle, cn } from 'util/index'
-
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue
+} from 'components/ui/select'
+import InfoButton from 'components/ui/info'
+import { useViewHeader } from 'contexts/ViewHeaderContext'
+import getMe from 'store/selectors/getMe'
+import getMyMemberships from 'store/selectors/getMyMemberships'
+import MembershipSettingsRow from './MembershipSettingRow'
+import SettingsToggles from './SettingToggles'
+
+import {
+  updateMembershipSettings,
+  updateAllMemberships,
+  updateUserSettings
+} from '../UserSettings.store'
+
+import { bgImageStyle } from 'util/index'
 import classes from './NotificationSettingsTab.module.scss'
 
 const allGroupsLogo = '/hylo-merkaba.png'
 
-const { object, func } = PropTypes
-
 const iOSAppURL = 'https://itunes.apple.com/app/appName/id1002185140'
 const androidAppURL = 'https://play.google.com/store/apps/details?id=com.hylo.hyloandroid'
 
-class NotificationSettingsTab extends Component {
-  static propTypes = {
-    currentUser: object,
-    updateUserSettings: func
-  }
+// Utility function to convert enum value to boolean flags
+const convertEnumToFlags = (value) => ({
+  sendEmail: includes(value, ['email', 'both']),
+  sendPushNotifications: includes(value, ['push', 'both'])
+})
+
+// Utility function to convert boolean flags to enum value
+const convertFlagsToEnum = ({ sendEmail, sendPushNotifications }) => {
+  if (sendEmail && sendPushNotifications) return 'both'
+  if (sendEmail) return 'email'
+  if (sendPushNotifications) return 'push'
+  return 'none'
+}
 
-  updateMessageSettings = changes => {
-    const { messageSettings, updateUserSettings } = this.props
-    const newMessageSettings = {
-      ...messageSettings,
+// Generic function to get current settings for any notification type
+const getCurrentSettings = (me, settingKey) => convertEnumToFlags(me.settings?.[settingKey])
+
+const getAllGroupsSettings = createSelector(
+  getMyMemberships,
+  memberships => ({
+    sendEmail: every(m => m.settings && m.settings.sendEmail, memberships),
```

Comment:
Not important, but curious if there was a reason you didn't use optional chaining (e.g. `m?.settings?.sendPushNotifications`) here and in the line below?

Reply:
just copying old code